openml-h2o/src/main/java/com/feedzai/openml/h2o/server/H2OApp.java

```diff
@@ -218,66 +188,31 @@ private Model train(final MLAlgorithmDescriptor algorithmDescriptor,
-                break;
-
-            case DEEP_LEARNING:
-                final DeepLearningV3.DeepLearningParametersV3 deepLearningParams = new H2ODeepLearningUtils().parseParams(trainingFrame, params, randomSeed, schema);
-
-                modelJob = train(() -> new DeepLearning(deepLearningParams.createAndFillImpl()).trainModel())
-                        .orElseThrow(() -> createModelTrainingException(algorithmDescriptor));
-                break;
 
-            case GRADIENT_BOOSTING_MACHINE:
-                final GBMV3.GBMParametersV3 gbmParams = new H2OGbmUtils().parseParams(trainingFrame, params, randomSeed, schema);
+        AbstractH2OAlgoUtils h2OAlgoUtils = H2OAlgoUtilsFactory.getH2OAlgoUtils(algorithmDescriptor, schema);
 
-                modelJob = train(() -> new GBM(gbmParams.createAndFillImpl()).trainModel())
-                        .orElseThrow(() -> createModelTrainingException(algorithmDescriptor));
-                break;
+        final Job<M> modelJob = train(() -> h2OAlgoUtils.train(trainingFrame, params, randomSeed, schema))
+                .orElseThrow(() -> createModelTrainingException(algorithmDescriptor));
 
-            case NAIVE_BAYES_CLASSIFIER:
-                final NaiveBayesV3.NaiveBayesParametersV3 naiveBayesParams = new H2OBayesUtils().parseParams(trainingFrame, params, randomSeed, schema);
-                modelJob = train(() -> new NaiveBayes(naiveBayesParams.createAndFillImpl()).trainModel())
-                        .orElseThrow(() -> createModelTrainingException(algorithmDescriptor));
-                break;
-
-            case GENERALIZED_LINEAR_MODEL:
-                final GLMV3.GLMParametersV3 glmParams = new H2OGeneralizedLinearModelUtils(schema).parseParams(trainingFrame, params, randomSeed, schema);
-
-                modelJob = train(() -> new GLM(glmParams.createAndFillImpl()).trainModel())
-                        .orElseThrow(() -> createModelTrainingException(algorithmDescriptor));
-                break;
-
-            case ISOLATION_FOREST:
-                final IsolationForestV3.IsolationForestParametersV3 isolationForestParams = new H2OIsolationForestUtils().parseParams(trainingFrame, params, randomSeed, schema);
+        return waitForModelTrained(modelJob);
+    }
 
-                modelJob = train(() -> new IsolationForest(isolationForestParams.createAndFillImpl()).trainModel())
-                        .orElseThrow(() -> createModelTrainingException(algorithmDescriptor));
-                break;
+    /**
+     * Validates the specified algorithm in the H2O platform.
+     * @param algorithmDescriptor The algorithm to train.
+     * @param schema              The schema for the model to be trained.
+     * @param params              The algorithm params.
+     *
+     * @since @@@feedzai.next.release@@@
+     */
+    public void validate(final MLAlgorithmDescriptor algorithmDescriptor,
+                                  final DatasetSchema schema,
```

Comment:
use the same indentation


this method should return a `ParamValidationError`